src/compile/model.ts

```diff
@@ -260,10 +274,17 @@ export abstract class Model {
    * Return the data source name for the given data source type.
    *
    * For unit spec, this is always simply the spec.name + '-' + dataSourceType.
-   * We already use the name map so that marks and scales use the correct data.
+   * The data name has not been renamed yet. Call `dataName` to get the final name.
    */
   public dataName(dataSourceType: DataTable): string {
-    return this._dataNameMap.get(this.name(String(dataSourceType)));
+    return this.name(String(dataSourceType));
```

Comment:
why `String(...)`, not `... as string`?  Our enum are `string` already so it seems unnecessary to call the function again.